hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -303,7 +305,7 @@ protected void commit(Option<Map<String, String>> extraMetadata, HoodieWriteMeta
   @SuppressWarnings("unchecked")
   protected Iterator<List<WriteStatus>> handleUpsertPartition(String instantTime, Integer partition, Iterator recordItr,
                                                               Partitioner partitioner) {
-    UpsertPartitioner upsertPartitioner = (UpsertPartitioner) partitioner;
+    SparkHoodiePartitioner upsertPartitioner = (SparkHoodiePartitioner) partitioner;
```

Comment:
is this renamed?

Reply:
Introduce a `SparkHoodiePartitioner` to implements both spark partitioner and hudi partitioner which is to get bucket info. The `UpsertPartitioner` inherit it.